jetty-core/jetty-websocket/jetty-websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandlerMetadata.java

```diff
@@ -46,6 +46,7 @@ public void setAutoDemand(boolean autoDemand)
     public void setBinaryHandle(Class<? extends MessageSink> sinkClass, MethodHandle binary, Object origin)
     {
         assertNotSet(this.binaryHandle, "BINARY Handler", origin);
+        assertNotSet(this.frameHandle, "FRAME Handler", origin);
```

Comment:
This is a very late thought bubble.... a frame handler can be for a signatures of:
```
 void <methodName>(Frame frame);
 void <methodName>(Session session, Frame frame);
```

Is it true, that it is only the second signature that is a problem, as they can try to use the session to do demands and send responses etc.

Perhaps we should allow the first signature, as that would be the case of somebody just "listening" to frames destined to other handlers???

Reply:
@gregw I don't think we need to be so strict because applications would be able to get a `Session` reference in other ways, so passing it here is not any more dangerous than passing it elsewhere.

Reply:
OK.   Was just trying to think of a way to allow the use-case that @lachlan-roberts highlighted of just monitoring the frames, but handling them in the other methods.

Reply:
The `Session.Listener` only makes the `Session` available from `onWebSocketOpen` anyway, so its quite typical to just save the session as a field, so doesn't make much difference if it is in the signature or not.